Adaptors/Amqp/src/QueueStorage.cs

```diff
@@ -182,6 +183,12 @@ public ValueTask<bool> Check(HealthCheckTag tag)
                      priority / MaxInternalQueuePriority,
                      priority % MaxInternalQueuePriority);
 
+    if (priority > MaxPriority || priority > MaxInternalQueuePriority )
```

Comment:
`priority` can be larger than `MaxInternalQueuePriority`. This the sole purpose of multiple queues here.

Reply:
Without the exception we run into a index out bounds if a user inserts a `priority`  larger than `MaxInternalQueuePriority`, I think the idea is that in such case, we interpret that the message is reaaaally important for the user and put it in the last queue. I edited the code to reproduce that.